docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+|----------|--------------------------------------------------|-----------------------------------|
+| id       | snowflake                                        | the ID of the invoked command     |
+| name     | string                                           | the name of the invoked command   |
+| options? | array of ApplicationCommandInteractionDataOption | the params + values from the user |
+
+###### ApplicationCommandInteractionDataOption
+
+All options have names, and an option can either be a parameter and input value--in which case `value` will be set--or it can denote a subcommand or group--in which case it will contain a top-level key and another array of `options`. 
+
+`value` and `options` are mututally exclusive.
+
+| Field    | Type                                             | Description                                     |
+|----------|--------------------------------------------------|-------------------------------------------------|
+| name     | string                                           | the name of the parameter                       |
+| value?   | OptionType                                       | the value of the pair                           |
+| options? | array of ApplicationCommandInteractionDataOption | present if this option is a group or subcommand |
+
+###### Interaction Response
+
+After receiving an interaction, you must respond to acknowledge it. This may be a `pong` for a `ping`, a message, or simply an acknowledgement that you have received it and will handle the command async.
+
+Interaction responses may choose to "eat" the user's command input if you do not wish to have their slash command show up as message in chat. This may be helpful for slash commands, or commands whose responses are asynchronous or ephemeral messages.
+
+| Field | Type                                      | Description                  |
+|-------|-------------------------------------------|------------------------------|
+| type  | InteractionResponseType                   | the type of response         |
+| data? | InteractionApplicationCommandCallbackData | an optional response message |
+
+###### InteractionResponseType
+
+| Name                     | Value | Description                                                       |
+|--------------------------|-------|-------------------------------------------------------------------|
+| Pong                     | 1     | ACK a `Ping`                                                      |
+| Acknowledge              | 2     | ACK a command without sending a message, eating the user's input  |
+| ChannelMessage           | 3     | respond with a message, eating the user's input                   |
+| ChannelMessageWithSource | 4     | respond with a message, showing the user's input                  |
+| ACKWithSource            | 5     | ACK a command without sending a message, showing the user's input |
```

Comment:
For both `2` and `5`, if you provide `data` a message still gets sent, which is against the behaviour described here.

Also I would have found a bitflag more intuitive but too late to change that I guess. (`Flag.DisplaySource | Flag.SendMessage` would be `4`, for example. I did this in my repo "slash-worker")

Reply:
That sounds like a bug, we can investigate.